### PR TITLE
RssAgent: Migrate from FeedNormalizer to Feedjira

### PR DESCRIPTION
**Pull request by knu**
This replaces #957.

### CODE REVIEW

File: app/models/agents/rss_agent.rb [hunk 1]
```diff
@@ -139,21 +162,241 @@ def check_and_track(entry_id)
       end
     end
 
+    LINK_ATTRS = %i[href rel type hreflang title length]
+    ENCLOSURE_ATTRS = %i[url type length]
+
+    unless dependencies_missing?
+      class AtomAuthor
```
Comment (cantino) [inline app/models/agents/rss_agent.rb]: Should we pull this into a separate file?

File: app/models/agents/rss_agent.rb [hunk 2]
```diff
@@ -139,21 +162,241 @@ def check_and_track(entry_id)
...
+        authors: authors,
+        date_published: feed.published,
+        last_updated: feed.updated || feed.published,
+      }
+    end
+
+    def entry_data(entry)
+      description = entry.summary
+      content = entry.content || description
+      id = entry.try(:entry_id) || Digest::MD5.hexdigest(content || '')
```
Comment (cantino) [inline app/models/agents/rss_agent.rb]: Might be worth keeping `get_entry_id`, as #1553 modifies it.

File: spec/data_fixtures/onethingwell.atom [hunk 3]
```diff
@@ -208,8 +209,7 @@
         </item>
         <item>
             <title>Showgoers</title>
-            <description>&lt;a href="http://showgoers.tv/"&gt;Showgoers&lt;/a&gt;: &lt;blockquote&gt; &lt;p&gt;Showgoers is a Chrome browser extension to synchronize your Netflix player with someone else so that you can co-watch the same movie on different computers with no hassle. Syncing up your player is as easy as sharing a URL.&lt;/p&gt; &lt;/blockquote&gt;
-            </description>
+            <description>&lt;a href="http://showgoers.tv/" onmouseover="javascript:void(0)"&gt;Showgoers&lt;/a&gt;: &lt;blockquote&gt; &lt;p&gt;Showgoers is a Chrome browser extension to synchronize your Netflix player with someone else so that you can co-watch the same movie on different computers with no hassle. Syncing up your player is as easy as sharing a URL.&lt;/p&gt; &lt;/blockquote&gt;&lt;script&gt;some code&lt;/script&gt;</description>
```
Comment (cantino) [inline spec/data_fixtures/onethingwell.atom]: So it no longer removes onmouseover / script, but it still safely escapes?

Comment (cantino) [inline spec/data_fixtures/onethingwell.atom]: Oh, this is with clean = false.

**Comment by cantino** [comment]
Should we ask some of the people from #889 and #955 to test this?

File: app/models/agents/rss_agent.rb (hunk 1 above)
Comment (knu) [inline app/models/agents/rss_agent.rb]: Maybe.  Would `app/models/agents/rss_agent/feedjira_extension.rb` be a good place?

Comment (cantino) [inline app/models/agents/rss_agent.rb]: Or maybe in `lib/feedjira_extension.rb`?

File: app/models/agents/rss_agent.rb [hunk 4]
```diff
@@ -139,21 +160,80 @@ def check_and_track(entry_id)
       end
     end
 
+    unless dependencies_missing?
+      require ''
```
Comment (cantino) [inline app/models/agents/rss_agent.rb]: `require ''`?

**Comment by knu** [comment]
Oh, I didn't mean to push this yet.  I'll fix it later.

**Comment by cantino** [comment]
Gotcha.

**Comment by knu** [comment]
OK, I think this is finally getting ready for merge.  Since I've refactored and rewritten a large portion of what I pushed the last time, I've squashed everything into a single commit so the whole change gets reviewed again from the ground up.  Thank you for your review and testing!

File: app/models/agents/rss_agent.rb [hunk 5]
```diff
@@ -53,18 +52,43 @@ def default_options
       Events look like:
 
           {
+            "feed": {
```
Comment (cantino) [inline app/models/agents/rss_agent.rb]: Maybe

```
{
  "feed": {  # if you set `include_feed_info` to `true` 
```

File: lib/feedjira_extension.rb [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+    include SAXMachine
+
+    def content=(content)
+      super
+
+      begin
+        addr = Mail::Address.new(content)
+      rescue
+        self.name = content
+      else
```
Comment (cantino) [inline lib/feedjira_extension.rb]: Cool, I didn't know you could do this on a `begin` block.

**Comment by cantino** [comment]
This is a significant piece of work, thank you!

Want to ask folks in #889 and #955 to test this?

File: app/models/agents/rss_agent.rb (hunk 5 above)
Comment (knu) [inline app/models/agents/rss_agent.rb]: Actually there was a note at the bottom but it was probably unnoticeable.  I've added a couple of additional notes about some fields, so it has hopefully gained its presence.

**Comment by knu** [comment]
@cantino Yes, I'd much appreciate if people who have had trouble with the RssAgent could test this out and give some feedback.  /cc @marycanady, @schulzpin, @Jngai

**Comment by knu** [comment]
I've been running this with various feeds for two weeks without a problem.  I'm afraid I was too late to come up when people had already found out their way to work around their problems, but is there any feedback from you folks?

I'd like to merge this soon, to move away from obsolete libraries and add features based on the new code base.

**Comment by knu** [comment]
@cantino Is it about time we should merge this and move on?

**Comment by cantino** [comment]
Yes @knu, I agree. Go for it!

**Comment by knu** [comment]
Done!

**Comment by dsander** [comment]
Awesome work 👏

**Comment by knu** [comment]
Thanks.  Google Reader is long gone and it's been one year since Yahoo Pipes shut down, but feeds shall not die!

**Comment by mcanady** [comment]
Thanks for your work on this--I'm in "set it and forget it" mode for this project so I can't test now, but will let you know when I do!

**Pull request merged**